paddle/fluid/framework/new_executor/interpreter/dependency_builder.h

```diff
@@ -37,17 +23,44 @@ namespace paddle {
 namespace framework {
 namespace interpreter {
 
-// equivalent to add_reader_dependency_pass
-void AddDependencyForReadOp(
-    const std::vector<Instruction>& vec_instruction,
-    std::map<int, std::list<int>>* downstream_map,
-    const std::vector<std::vector<bool>>* op_happens_before = nullptr);
-
-void AddDownstreamOp(
-    int prior_op_idx,
-    int posterior_op_idx,
-    std::map<int, std::list<int>>* op_downstream_map,
-    const std::vector<std::vector<bool>>* op_happens_before = nullptr);
+// DependencyBuilder provides some dependency adding function to handle the
+// dependency that cannot be explicitly expresed by a Program. It is a
+// compromise of the incomplete expression ability of the Program. Do not add
+// too many functions here at will, that will bring great burden to the
+// Interpretercore.
+
+class DependencyBuilder {
+ public:
+  explicit DependencyBuilder() : instructions_(nullptr) {}
+
+  const std::map<int, std::set<int>>& Build(
```

Comment:
Better add comments on the function, especially the  `return`, or, rename it to specify that it returns `op_downstream_map_`

Reply:
Done